src/util/feefrac.h

```diff
@@ -144,6 +186,39 @@ struct FeeFrac
         std::swap(a.fee, b.fee);
         std::swap(a.size, b.size);
     }
+
+    /** Compute the fee for a given size `at_size` using this object's feerate.
+     *
+     * This effectively corresponds to evaluating (this->fee * at_size) / this->size, with the
+     * result rounded towards negative infinity (if RoundDown) or towards positive infinity
+     * (if !RoundDown).
+     *
+     * Requires this->size > 0, at_size >= 0, and that the correct result fits in a int64_t. This
+     * is guaranteed to be the case when 0 <= at_size <= this->size.
+     */
+    template<bool RoundDown>
+    int64_t Evaluate(int32_t at_size) const noexcept
+    {
+        Assume(size > 0);
+        Assume(at_size >= 0);
+        if (fee >= 0 && fee < 0x200000000) [[likely]] {
```

Comment:
Was curious how much difference using `[[likely]]` may/may-not have, as I've seen mixed discussions around it's usage. For a single call to `EvaluateDown` (from the profile_estimate pass):

![combined](https://github.com/user-attachments/assets/83eb899c-e4b2-4bff-a402-f4a45109a3f2)

Reply:
I personally view it as documentation and would definitely keep it here